app/main.py

```diff
@@ -1,3 +1,6 @@
+from app.battle import battle
+
+
 KNIGHTS = {
```

Comment:
You could place the `KNIGHTS` variable in a separate file, e.g. `knights_config.py`, and leave the `battle` function in this file.